Add right click context menu summarize feature

IcedRooibos opened the pull request:
Follow on from: https://github.com/kagisearch/kagi-docs/pull/199

Tested on Firefox 118.0.1 and Google Chrome 117.0.5938.132

BrunoBernardino commented on chrome/manifest.json:
```diff
@@ -22,7 +22,9 @@
-    "storage"
+    "storage",
+	"contextMenus",
+	"tabs"
```
@IcedRooibos is this permission necessary? We had it and reverted due to #13

IcedRooibos replied:
Good point, not something I had considered - I shall test this along with the other change over the weekend.

BrunoBernardino commented on chrome/manifest.json:
```diff
@@ -22,7 +22,9 @@
     "webRequest",
-    "storage"
+    "storage",
+	"contextMenus",
```
This indentation difference makes it look like you didn't run `npm run format`, so please confirm that and `npm test` passes.

IcedRooibos replied:
Hi there, good point - you are correct. I did indeed forget to run npm test.

There is another change that's needed based on npm test - the `let` variables should be `const`s.

I'll get to this over the weekend alternatively if you prefer to do it sooner I don't mind if someone from Kagi team does it either - all good either way.

Thanks for the guidance!

BrunoBernardino replied:
I'll leave you to it, it's not urgent. :blush:

BrunoBernardino reviewed (changes requested):
Thanks for this @IcedRooibos ! I haven't pulled this down yet, but have a question about the `tabs` permission, which we had to remove a while back, as it's too broad. Can this be used without it? Doesn't the `activeTab` work for it?

IcedRooibos commented:
@BrunoBernardino Just updated it, let me know if you think any more changes are needed, happy to take another look.

BrunoBernardino commented:
> @BrunoBernardino Just updated it, let me know if you think any more changes are needed, happy to take another look.

Thank @IcedRooibos ! I just tried this out and I can't see the context menu. Is there anything else I need to do after giving it all the permissions for the current tab? If you could record a short gif/video it might help me out. I tried Chrome and Firefox (latest).

I also noticed `npm run format` fixes some formatting issues, so you might want to run that.

IcedRooibos commented:
@BrunoBernardino thanks, I've run `npm run format now` - it made changes to other files as well but I've only committed the ones that I have touched in this commit. Here are the other files changed by `npm run format` that I didn't commit.

```
        modified:   build.js
        modified:   rome.json
        modified:   shared/src/lib/utils.js
        modified:   shared/src/popup.js
        modified:   shared/src/summarize_result.js
```

Here's a video showing show it works - when you right click a link it gives you the option to send it to summarizer. Should also work on youtube etc. Did this not work when you tried it?

https://drive.proton.me/urls/1HQTZXWDR8#NK30nmkFulh5

BrunoBernardino reviewed (approved):
Perfect, thank you so much for sharing that, I was expecting to right click on "any website" and see an option to summarize that, not just links. This makes sense and works as expected in Firefox and Chrome, though.

Thanks again!

IcedRooibos commented:
My pleasure @BrunoBernardino and thank you for your help improving the PR quality as well!

alabhyajindal commented:
Thanks for this PR! 

I recently started using Kagi. I use the Universal Summarizer sometimes when I am on a page and want to look at a summary before reading the whole article. 

I think it makes more sense to trigger the option for article summarization when right clicking on a page, instead of a link. I don't see many people wanting to summarize a link they have not visited.

An example:

https://github.com/kagisearch/browser_extensions/assets/52493077/b8befe21-f0cd-4159-ac24-5baec5d67c68

IcedRooibos commented:
For my personal use case I tend to do it with Youtube or news articles to check what's actually worth reading before reading it but I can absolutely see why you'd want it to be triggered from the page you're on as well. I think should be able to do both right?

BrunoBernardino commented:
I agree both would be great!

vprelovac commented:
@alabhyajindal @IcedRooibos agreed, please push an MR for this and we will add it. Awaiting this and review of Image search and we will push out new version of the extension. Great work everyone!
